src/js/components/Button/StyledButtonKind.js

```diff
@@ -78,8 +78,17 @@ const basicStyle = props => css`
   ${padStyle(props)}
   ${fontStyle(props)}
 
-  > svg {
+  // when button has badge, the SVG won't necessarily
+  // be the direct descendant
+  ${
+    props.badge
+      ? `
+  svg {
```

Comment:
What if the Button has an `<svg />` child? Would it be possible to put this inside a StyledBadge?

Reply:
Good thought, let me play around with that.

Reply:
Following on here, I think this adjustment should be fine and it makes sense for it to live in `basicStyle`. The intent is that the selector applies to an svg inside the Button. When user does the following, the svg is a direct descendant of the Button, so the original `> svg` selector works:
```
<Button icon={<Add />} />
```

However, when the user adds badge, Stack is introduced into the Button contents, meaning that the svg is no longer the direct descendant of Button, so we need to adjust the selector:
```
<Button icon={<Add />} badge={2} />
```
This is affecting `basicStyle` meaning it will apply when the user is relying on Button props like `icon` and `label`. However, it will take no effect on something where caller has chosen to provide children to Button (because providing children to Button results in the button using `plainStyle`. See below for rendering of the following code:
```
<Button badge={2}>
   <Box background="skyblue" height="xsmall" width="xsmall">
      <MailOption />
  </Box>
</Button>
```
<img width="138" alt="Screen Shot 2021-05-05 at 11 50 30 AM" src="https://user-images.githubusercontent.com/12522275/117194346-dbd90800-ad98-11eb-931f-2aaecc032479.png">